### ISSUE TITLE
TestLogic/fakedist-vec-off/schema_change_in_txn/rollback_mutations test failure

### ISSUE THREAD

**Issue by stevendanna** [cockroachdb/cockroach#83682]
I've seen this on a couple of PRs:

- https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_GitHubCiOptional/5614915?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true
- https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_GitHubCiOptional/5615157?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
=== RUN   TestLogic/fakedist-vec-off/schema_change_in_txn/rollback_mutations
[20:40:30] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8473/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/sequences: 299 statements
=== CONT  TestLogic/fakedist-vec-off/schema_change_in_txn/rollback_mutations
    logic.go:2937:
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8473/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn:726:
        expected:
        violates unique constraint
        got:
        pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh due to a conflict: committed value on key /Table/15/1/775131069132046337/2/1): "sql txn" meta={id=add35662 key=/Table/3/1/138/2/1 pri=0.00620047 epo=0 ts=1656621631.138192625,1 min=1656621630.762259669,0 seq=67} lock=true stat=PENDING rts=1656621630.762259669,0 wto=false gul=1656621631.262259669,0
        HINT: See: https://www.cockroachlabs.com/docs/v22.2/transaction-retry-error-reference.html#retry_serializable
    logic.go:2577:
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8473/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn:729: too many errors encountered, skipping the rest of the input
[20:40:31] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8473/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn with config fakedist-vec-off: 191 tests, 2 failures
[20:40:31] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8473/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema: 69 statements
            --- FAIL: TestLogic/fakedist-vec-off/schema_change_in_txn/rollback_mutations (1.09s)
```

Jira issue: CRDB-17206

**Comment by ajwerner** [comment]
https://github.com/cockroachdb/cockroach/issues/52366 has been our tracking issue for this.

**Comment by rafiss** [comment]
resolved by 28c48f9b5d8eade07934c7a894a98aa5a790347d
